src/app/tools/controllers.h

```diff
@@ -17,6 +17,94 @@ namespace app { namespace tools {
 
 using namespace gfx;
 
+// Adjustment for snap to isometric grid
+static void snap_isometric_line(ToolLoop* loop, Stroke& stroke, bool lineCtl)
+{
+  // Get last two points
+  Stroke::Pt& a = stroke[stroke.size() - 2];
+  Stroke::Pt& b = stroke[stroke.size() - 1];
+
+  // Get function invoked by line tool
+  bool lineTool = (string_id_to_brush_type(loop->getTool()->getId()) == kLineBrushType);
+
+  // TODO: rectangles and ellipses
+  if (lineCtl && !loop->getIntertwine()->snapByAngle())
+    return;
+
+  // Get line angle
+  PointF vto(b.x - a.x, b.y - a.y);
+  double len = ABS(vto.x) + ABS(vto.y);
+  vto /= len;
+
+  const gfx::Rect& grid = loop->getGridBounds();
+  const auto line = doc::Grid::IsometricGuide(grid.size());
+
+  // Offset vertical lines/single point to the left for line tool.
+  // Because pressing the angle snap key will bypass this function,
+  // this makes it so one can selectively apply the offset.
+  if ((std::isnan(vto.x) && std::isnan(vto.y)) || (int(vto.x) == 0 && int(vto.y) != 0)) {
+    const int step = 1 + (line.oddSize * int(!line.squareRatio));
+    a.x -= step * int(lineTool);
+    b.x -= step * int(lineTool);
+  }
+  // Horizontal lines
+  else if (int(vto.y) == 0 && int(vto.x) != 0) {
+    if (vto.x > 0)
+      b.x--;
+    else
+      a.x--;
+  }
+  // Diagonal lines
+  else {
+    // Adjust start/end point based on line direction and grid size
+    if (!line.squareRatio) {
+      if (vto.x < 0) {
+        a.x -= line.evenWidth;
+        b.x -= 2 * line.oddSize;
+      }
+      else {
+        a.x -= 2 * line.oddSize;
+        b.x -= line.evenWidth;
+      }
+
+      // Unticking 'share borders' adds one pixel of distance between edges
+      if (!line.shareEdges) {
+        if (vto.y < 0)
+          a.y--;
+        else
+          b.y--;
+      }
+
+      // Some line angles do not intertwine in the exact same way
+      // when the order of the two points is inverted, so we try to
+      // detect this edge case and flip the points.
+      //
+      // TODO: this fix only works for two-point lines. Support
+      // for freehand strokes would require changes to intertwiners,
+      // not just the freehand controller itself.
+      if (lineTool && vto.x < 0 && a.x % (grid.w - !line.evenWidth)) {
+        auto tmp = a;
+        a = b;
+        b = tmp;
```

Comment:
You can use `std::swap()` (needs `<utility>` probably)
```suggestion
        std::swap(a, b);
```